ta/crypt/cryp_taf.c

```diff
@@ -47,104 +49,121 @@ TEE_Result ta_entry_allocate_operation(uint32_t param_type, TEE_Param params[4])
 	res = TEE_AllocateOperation(&op,
 				    params[0].value.b, params[1].value.a,
 				    params[1].value.b);
-	params[0].value.a = (uint32_t) op;
+	params[0].value.a = (uintptr_t)op;
```

Comment:
`a` is still `uint32_t`, isn't it? Then what is the guarantee that `op` isn't larger?